sys/bitfield/bitfield.c

```diff
@@ -46,3 +46,27 @@ int bf_get_unset(uint8_t field[], int size)
     restoreIRQ(state);
     return(result);
 }
+
+int bf_get_first_set(uint8_t field[], int size)
+{
+    int result = -1;
+    int nbytes = (size + 7) / 8;
+    int i = 0;
+
+    unsigned state = disableIRQ();
+
+    /* skip full bytes */
```

Comment:
skipping words until a positive match is found and then skipping bytes might converge faster